Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/frame/DartVmServiceEvaluatorOnLibrary.java

```diff
@@ -0,0 +1,117 @@
+package com.jetbrains.lang.dart.ide.runner.server.vmService.frame;
+
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import com.intellij.xdebugger.XSourcePosition;
+import com.intellij.xdebugger.evaluation.ExpressionInfo;
+import com.intellij.xdebugger.evaluation.XDebuggerEvaluator;
+import com.jetbrains.lang.dart.ide.runner.server.vmService.DartVmServiceDebugProcess;
+import com.jetbrains.lang.dart.psi.DartFile;
+import com.jetbrains.lang.dart.psi.DartPartOfStatement;
+import gnu.trove.THashSet;
+import org.dartlang.vm.service.element.Isolate;
+import org.dartlang.vm.service.element.LibraryRef;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+
+public class DartVmServiceEvaluatorOnLibrary extends XDebuggerEvaluator {
+  @NotNull private final DartVmServiceDebugProcess myDebugProcess;
+  @NotNull private final String myIsolateId;
+  @NotNull private final VirtualFile myFile;
+  @NotNull private final Project myProject;
+
+  public DartVmServiceEvaluatorOnLibrary(@NotNull final DartVmServiceDebugProcess debugProcess,
+                                         @NotNull final String isolateId,
+                                         @NotNull final VirtualFile file,
+                                         @NotNull final Project project) {
+    myDebugProcess = debugProcess;
+    myIsolateId = isolateId;
+    myFile = file;
+    myProject = project;
+  }
+
+  @Override
+  public boolean isCodeFragmentEvaluationSupported() {
+    return true;
+  }
+
+  @Override
+  public void evaluate(@NotNull final String expression,
+                       @NotNull final XEvaluationCallback callback,
+                       @Nullable final XSourcePosition expressionPosition) {
```

Comment:
If you Alt+click or select and Alt+click something in the editor then `expressionPosition` is not `null`. Although it is probably the same as `myFile`, I think it would be more accurate to use it explicitly.